harvester_e2e_tests/integration/test_vm_functions.py

```diff
@@ -30,6 +37,51 @@ def kubeconfig_file(api_client):
         yield Path(f.name)
 
 
+@pytest.fixture(scope='module')
+def bogus_vlan_net(request, api_client):
+    """bogus vlan network fixture (no dhcp) on mgmt network
+
+    Args:
+        request (FixtureRequest): https://docs.pytest.org/en/7.1.x/_modules/_pytest/fixtures.html#FixtureRequest # noqa
+        api_client (HarvesterAPI): HarvesterAPI client
+
+    Yields:
+        dict: created bogus network attachment definition dictionary
+    """
+    original_vlan_id = request.config.getoption('--vlan-id')
+
+    existing_vm_net_code, existing_vm_net_data = api_client.networks.get()
+    assert existing_vm_net_code == 200, 'we should be able to fetch vm networks from harvester'
+    existing_vm_net_list = existing_vm_net_data.get('items', [])
+    vlans_to_exclude = set()
+    vlans_to_exclude.add(1)
+    for existing_vm_net in existing_vm_net_list:
+        existing_vm_net_config = existing_vm_net.get('spec', {}).get('config', '{}')
+        assert existing_vm_net_config != '{}', 'existing vm net should exist'
+        existing_vm_net_config_dict = json.loads(existing_vm_net_config)
+        assert existing_vm_net_config_dict.get(
+            'vlan', 0) != 0, 'we should be able to get the vlan off the config'
+        existing_vm_net_vlan = existing_vm_net_config_dict.get('vlan')
+        vlans_to_exclude.add(existing_vm_net_vlan)
+
+    if original_vlan_id != -1:
+        vlans_to_exclude.add(original_vlan_id)
+
+    # create a random vlan that 'shouldnt' have dhcp
+    # create based on excluding some values and from within a range as a set operation
+    vlan_id = choice(list(set([vlan for vlan in range(1, 4094)]) - vlans_to_exclude))
+
+    vm_network_name = f'bogus-net-{vlan_id}'
+    code, data = api_client.networks.create(vm_network_name, vlan_id)
+    assert 201 == code, (
+        f"Failed to create N.A.D. {vm_network_name} with error {code}, {data}"
+    )
+
+    yield data
```

Comment:
if we will just need `data['metadata']['namespace']` and `data['metadata']['name']`, we can only yield them out.

Reply:
It might be nice to yield back the whole complete object from that fixture - if in the future more checks need to utilize other properties.

We could possibly build another dict with those elements and yield that to trim things back, like:
```python
vlan_info = {
  "namespace": data['metadata']['namespace'],
  "name": data['metadata']['name'],
}
yield vlan_info
```

I was thinking it might just be easier to yield back `data` entirely. 
So I left it for now - but I'm open to change that too :smile: